test/RelayHubGasCalculations.test.ts

```diff
@@ -184,25 +197,25 @@ contract('RelayHub gas calculations', async function ([_, relayOwner, relayAddre
     }
   }
 
-  function calculateOverchargeForCalldata (relayRequest, signature) {
+  function calculateOverchargeForCalldata (relayRequest: RelayRequest, signature: string): BN {
     const calldata = relayHub.contract.methods.relayCall(relayRequest, signature, '0x').encodeABI()
     const calldataSize = calldata.length / 2 - 1
     const calldataBuffer = Buffer.from(calldata.slice(2), 'hex')
-    const correctGasCost1 = correctGasCost(calldataBuffer, gtxdatanonzero, gtxdatazero, 3)
+    const correctGasCost1 = correctGasCost(calldataBuffer, gtxdatanonzero, gtxdatazero)
     return new BN(calldataSize * gtxdatanonzero - correctGasCost1).mul(gasPrice)
   }
 
-  function logOverhead (weiActualCharge, overchargeForCalldata, weiGasUsed) {
+  function logOverhead (weiActualCharge: BN, overchargeForCalldata: BN, weiGasUsed: BN): void {
     const actualChargeWithoutOvercharge = weiActualCharge.sub(overchargeForCalldata)
     const gasDiff = weiGasUsed.sub(actualChargeWithoutOvercharge).div(gasPrice).toString()
     if (gasDiff !== '0') {
       console.log('== zero-fee unmatched gas. RelayHub.gasOverhead should be increased by: ' + gasDiff.toString())
     }
   }
 
-  describe('check calculation does not for different fees', async function () {
+  describe('check calculation does not break for different fees', function () {
     before(async function () {
-      await relayHub.depositFor(relayOwner, { value: 1 })
+      await relayHub.depositFor(relayOwner, { value: (1).toString() })
```

Comment:
strange. can't this be a number? (or at least BN)

Reply:
value is `string|BN`, and that is actually smart considering how badly `number` holds up for big values